specification/containerservice/Fleet.Management/update/autoupgradeprofile.tsp

```diff
@@ -0,0 +1,119 @@
+import "@typespec/rest";
+import "@typespec/openapi";
+import "@typespec/versioning";
+import "@azure-tools/typespec-providerhub";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "./../helpers.tsp";
+import "./run.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Foundations;
+using Azure.Core;
+using Azure.Core.Traits;
+using TypeSpec.OpenAPI;
+
+namespace Microsoft.ContainerService;
+
+@resource("autoUpgradeProfiles")
+@parentResource(Fleet)
+@added(Versions.v2024_05_02_preview)
+@doc("The AutoUpgradeProfile resource.")
+model AutoUpgradeProfile is ProxyResource<AutoUpgradeProfileProperties> {
+  @doc("The name of the AutoUpgradeProfile resource.")
+  @key("autoUpgradeProfileName")
+  @segment("autoUpgradeProfiles")
+  @pattern("^[a-z0-9]([-a-z0-9]*[a-z0-9])?$")
+  @minLength(1)
+  @maxLength(50)
+  @path
+  @visibility("create", "read")
+  name: string;
+
+  ...EntityTag;
+}
+
+@doc("The provisioning state of the AutoUpgradeProfile resource.")
+enum AutoUpgradeProfileProvisioningState {
+  ...ResourceProvisioningState,
+}
+
+@doc("The properties of the AutoUpgradeProfile.")
+model AutoUpgradeProfileProperties {
+  @visibility("read")
+  @doc("The provisioning state of the AutoUpgradeProfile resource.")
+  provisioningState?: AutoUpgradeProfileProvisioningState;
+
+  @visibility("create", "read")
+  @doc("The resource id of the UpdateStrategy resource to reference. If not specified, the auto upgrade will run on all clusters which are members of the fleet.")
+  updateStrategyId?: FleetUpdateStrategyResourceId;
+
+  @visibility("create", "read")
+  @doc("Configures how auto-upgrade will be run.")
+  channel: UpgradeChannel;
+
+  @visibility("create", "read")
+  @doc("The node image upgrade to be applied to the target clusters in auto upgrade.")
+  @added(Versions.v2024_05_02_preview)
```

Comment:
why explicitly specify the version here since the entire resource already has `@added(Versions.v2024_05_02_preview)`

Reply:
That doesn't make model AutoUpgradeProfileProperties versioned. Made the whole model versioned since that makes more sense that just this field